mongodb/src/main/java/org/hibernate/ogm/datastore/mongodb/MongoDBDialect.java

```diff
@@ -154,6 +155,44 @@ public MongoDBDialect(MongoDBDatastoreProvider provider) {
 	@Override
 	public Tuple getTuple(EntityKey key, TupleContext tupleContext) {
 		DBObject found = this.getObject( key, tupleContext );
+		return createTuple( key, tupleContext, found );
+	}
+
+	@Override
+	public List<Tuple> getTuples(EntityKey[] keys, TupleContext tupleContext) {
+		DBCursor cursor = this.getObjects( keys, tupleContext );
+		List<Tuple> tuples = createListOfTuples( keys );
+		try {
+			for ( DBObject dbObject : cursor ) {
+				for ( int i = 0; i < keys.length; i++ ) {
+					if ( matches( dbObject, keys[i].getColumnNames(), keys[i].getColumnValues() ) ) {
+						tuples.set( i, createTuple( keys[i], tupleContext, dbObject ) );
+					}
+				}
+			}
+			return tuples;
+		}
+		finally {
+			if ( cursor != null ) {
+				cursor.close();
+			}
+		}
+	}
+
+	private List<Tuple> createListOfTuples(EntityKey[] keys) {
+		List<Tuple> tuples = new ArrayList<>( keys.length );
+		for ( int i = 0; i < keys.length; i++ ) {
```

Comment:
> Ok, got it now. I think I'd rather add the nulls when iterating the cursor,

The number of results returned by the cursor might not match the number of keys, some entities might not be in the db. Not sure how you would do that.

> ... to give a less style-based argument 

Not sure what you mean